vhost_user_backend/src/lib.rs

```diff
@@ -537,17 +562,21 @@ impl<S: VhostUserBackend> VhostUserHandler<S> {
                 exit_event_id,
             };
 
-            worker_threads.push(
-                thread::Builder::new()
-                    .name("vring_worker".to_string())
-                    .spawn(move || vring_worker.run(vring_handler))
-                    .map_err(VhostUserHandlerError::SpawnVringWorker)?,
-            );
+            if VirtioDeviceType::from(backend.read().unwrap().get_type())
```

Comment:
Wrong way to do it, instead query the backend for it's workers and use them if provided.

Reply:
Remove the type query, but we could not get the worker_thread from backends at this moment, since it will be created after memory is update.